### PR TITLE
[Ready]You can now select what your pills will look like when making pills from the Chem Master

### PR DESCRIPTION
**Pull request by ghost**
:cl: Floyd/Qustinnus, paid for by Kryson
add: You can now select what your pills will look like when making pills from the Chem Master
/:cl:

we had 21 pill sprites and used 1. waste. now you can pick from our list of 21 and it'll show you beforehand what they look like

https://streamable.com/445u8

### CODE REVIEW

**Comment by Qustinnus** [comment]
updates so they now have names so oranges doesnt string me up

**Comment by ExcessiveUseOfCobblestone** [comment]
LOUIS!

**Comment by coiax** [comment]
At the risk of making you cry, could you try to make the "choose pill style" a seperate tgui menu which has all the options shown?

**Comment by Qustinnus** [comment]
i might make it so it just puts all the pills on the chem master menu. seperate interface sounds like hell and idk how to do it but I might try.

**Comment by nicbn** [comment]
Can you please make it so that there are right and left buttons to change it instead of having the popup selection window?

**Comment by Qustinnus** [comment]
sounds equally annoying due to there being 21 options so it mgiht give people pain in their finger especially seeing as its tgui and will probably lag

**Comment by nicbn** [comment]
Make it show a spoiler-like section then, please.
Spawning a new tgui window is quite an overkill and adds one extra window to player (who probably already has both the chem master and chem dispenser windows open).

**Comment by Qustinnus** [comment]
this is currently not opening another tgui menu, can you explain what you mean with a spoiler like section?

**Comment by nicbn** [comment]
A show/hide button that makes a section with the options appear

**Comment by Shadowflame909** [comment]
what kind of chemicals are we putting in these pills to change their color!

THEY COULD SEEP INTO THE WATER

**Comment by davethwave** [comment]
Food coloring?

**Comment by kriskog** [comment]
Very nice! 
Can you give them descriptive names? Think "Round Green" or "Red" or whatever they are.

**Comment by Qustinnus** [comment]
im colorblind so can you make a suggesed change for all names ud like?

File: code/modules/reagents/chemistry/machinery/chem_master.dm [hunk 1]
```diff
@@ -244,7 +251,51 @@
 				P.desc = "A small condiment pack. The label says it contains [name]."
 				reagents.trans_to(P,10, transfered_by = usr)
 			. = TRUE
-
+		if("pillStyle")
+			var/pill_name = input(usr, "Choose your desired pill style:", "Pill Style")  as null|anything in list("capsule 1", "capsule 2", "capsule 3", "capsule 4", "capsule 5", "capsule 6", "capsule 7", "capsule 8", "capsule 9", "capsule 10", "tablet 1", "tablet 2", "tablet 3", "tablet 4", "tablet 5", "tablet 6", "tablet 7", "tablet 8", "tablet 9", "tablet 10", "tablet 11")
+			switch(pill_name)
+				if("capsule1")
```
Comment (ShizCalev) [inline code/modules/reagents/chemistry/machinery/chem_master.dm]: don't these all have spaces?

**Comment by Razharas** [comment]
Choosing red colour should give you a message about wondering why lizard people make up 20% of station population but comitt 80% of the crime

File: code/modules/reagents/chemistry/machinery/chem_master.dm [hunk 2]
```diff
@@ -244,7 +251,51 @@
...
+				if("capsule 6")
+					pill_style = 6
+				if("capsule 7")
+					pill_style = 18
+				if("capsule 8")
+					pill_style = 19
+				if("capsule 9")
+					pill_style = 20
+				if("capsule 10")
+					pill_style = 20
```
Comment (Mickyan) [inline code/modules/reagents/chemistry/machinery/chem_master.dm]: duplicate? also missing styles 21 and 22

Comment (Qustinnus) [reply code/modules/reagents/chemistry/machinery/chem_master.dm]: 22 isnt used

Comment (Qustinnus) [reply code/modules/reagents/chemistry/machinery/chem_master.dm]: rather; its not supposed to be

**Comment by MrDoomBringer** [comment]
image
the menu now looks like this ur welcome

**Comment by Qustinnus** [comment]
image
thank mrdoombringer for being my best girlfriend ever

**Comment by Ziiro** [comment]
Based and Floorpilled

**Comment by PKPenguin321** [comment]
Option for random?

**Comment by Qustinnus** [comment]
smh its never enough with you people

jk ill look into it

**Comment by epochayur** [comment]
based doombringer

**Comment by MACIEKBAKI** [comment]
now i can make cyanide pills that look like cyanide pills at my local chem master!

**Comment by davethwave** [comment]
This will make the chem/medical traitor roll more interesting

File: code/modules/reagents/chemistry/machinery/chem_master.dm [hunk 3]
```diff
@@ -13,6 +13,7 @@
 	var/obj/item/storage/pill_bottle/bottle = null
 	var/mode = 1
 	var/condi = FALSE
+	var/list/chosenPillStyle = 1
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/machinery/chem_master.dm]: stop it, get some help

File: code/modules/reagents/chemistry/machinery/chem_master.dm [hunk 4]
```diff
@@ -156,7 +160,12 @@
 		for(var/datum/reagent/N in reagents.reagent_list)
 			bufferContents.Add(list(list("name" = N.name, "id" = N.id, "volume" = N.volume))) // ^
 		data["bufferContents"] = bufferContents
-
+	var/list/pillStyles = list()
+	for (var/x in 1 to 22)
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/machinery/chem_master.dm]: Magic number

File: code/modules/reagents/chemistry/machinery/chem_master.dm [hunk 5]
```diff
@@ -231,6 +240,12 @@
 					else
 						P = new(drop_location())
 					P.name = trim("[name] pill")
+					if(chosenPillStyle == 22)
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/machinery/chem_master.dm]: ditto

File: code/modules/reagents/chemistry/machinery/chem_master.dm [hunk 6]
```diff
@@ -231,6 +240,12 @@
...
+						P.icon_state ="pill[rand(1,21)]"
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/machinery/chem_master.dm]: Is this even correct?

File: code/modules/reagents/chemistry/machinery/chem_master.dm [hunk 7]
```diff
@@ -231,6 +240,12 @@
...
+					else
+						P.icon_state = "pill[chosenPillStyle]"
+					if(P.icon_state == "pill4")
+						P.desc = "A tablet or capsule, but not just any, a red one, one taken by the ones not scared of knowledge, freedom, uncertainty and the brutal truths of reality."
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/machinery/chem_master.dm]: -_-

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/machinery/chem_master.dm (hunk 6 above)
Comment (Qustinnus) [reply code/modules/reagents/chemistry/machinery/chem_master.dm]: yes

File: code/modules/reagents/chemistry/machinery/chem_master.dm (hunk 7 above)
Comment (Qustinnus) [reply code/modules/reagents/chemistry/machinery/chem_master.dm]: :D

File: code/modules/reagents/chemistry/machinery/chem_master.dm (hunk 3 above)
Comment (Qustinnus) [reply code/modules/reagents/chemistry/machinery/chem_master.dm]: blame @MrDoomBringer

Comment (MrDoomBringer) [reply code/modules/reagents/chemistry/machinery/chem_master.dm]: blame @Qustinnus for not checking the code i sent him :^P

File: code/modules/reagents/chemistry/machinery/chem_master.dm (hunk 7 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/machinery/chem_master.dm]: a round and full description

**Comment by davethwave** [comment]
YAY

**Pull request merged**